### ISSUE TITLE
detecting only one object : one class + blank false images VS multiple classs + non blank false images

### ISSUE THREAD

**Issue by HanSeYoung** [AlexeyAB/darknet#2381]
I'm a writer of this [question](https://github.com/AlexeyAB/darknet/issues/2366#issue-407983989)

What I really want is detecting just one object.

I wonder which scenario is better at finding one class object.

Situation : I wanna find a monkey in forest but some branches and leaves are detected as monkey.

> 1. Do not label branches and leaves. Just leave those images with blanked txt-file and label only monkeys.

> 2.  Do label branches as 'branch' and leaves as 'leaf', without blank background images.

> 3. Same with 'scenario 2' but add blank txt-file which is not among the branches, leaves and monkeys.

Thanks for reading my question!

**Comment by PeterQuinn925** [comment]
I had trouble with a similar scenario  - detecting squirrels. If I were you, I would take the false positives, the leaves that were detected as monkeys, and put them unlabeled in your training set. I would not add classes for leaf and branch.

I had more problems with false negatives than false positives and manually paged through test footage to find squirrels that Yolo had missed. I retrained several times. I think my biggest problem was camera resolution. I may set up the squirrel cam again this summer with a better camera and see how that goes.

**Comment by HanSeYoung** [comment]
@PeterQuinn925 Thanks for sharing your experience.

Did you test that scenario which you suggest to me?

Should I make branch and leaf labels with blank txt file?
Or don't make any label of them and just label background which monkey was not detected with blank txt file?

**Comment by AlexeyAB** [comment]
@HanSeYoung Hi,

The best scenario is:
1. Do not label branches and leaves. Just leave those images with blanked txt-file and label only monkeys.

Just make training dataset as big as possible.

**Comment by HanSeYoung** [comment]
@AlexeyAB 
I really appreciate with your answer!!

More detail question

Then just let the branches and leaves with background original image? or crop them from original image and save them each other? with blank txt-files?

**Comment by AlexeyAB** [comment]
@HanSeYoung 

As you want, but:

* If there are monkeys on this image - then every monkey must be labeled (txt-file isn't empty)

* If there are no monkeys on this image - then use empty txt-file.


Also if you crop them from original image, then you should keep the same **relative** scale of branches and leaves as it will be in the test images.
